src/amo/components/InstallButtonWrapper/index.js

```diff
@@ -99,32 +101,43 @@ export const InstallButtonWrapperBase = (props: InternalProps) => {
           'InstallButtonWrapper--notFirefox': !isFirefox({ userAgentInfo }),
         })}
       >
-        <AMInstallButton
-          addon={addon}
-          canUninstall={canUninstall}
-          className={makeClassName(
-            className ? `AMInstallButton--${className}` : '',
-            {
-              'AMInstallButton--noDownloadLink': !showDownloadLink,
-            },
-          )}
-          currentVersion={currentVersion}
-          defaultButtonText={defaultButtonText}
-          disabled={!isCompatible}
-          enable={enable}
-          hasAddonManager={hasAddonManager}
-          install={install}
-          isAddonEnabled={isAddonEnabled}
-          puffy={puffy}
-          setCurrentStatus={setCurrentStatus}
-          status={installStatus}
-          uninstall={uninstall}
-        />
-        <GetFirefoxButton
-          addon={addon}
-          buttonType={getFirefoxButtonType}
-          className={className ? `GetFirefoxButton--${className}` : ''}
-        />
+        {showLinkInsteadOfButton && (
+          <a className={makeClassName('oldDownload')} href={installURL}>
+            {i18n.gettext('Download Older Version')}
+          </a>
+        )}
+
+        {!showLinkInsteadOfButton && (
+          <>
+            <AMInstallButton
+              addon={addon}
+              canUninstall={canUninstall}
+              className={makeClassName(
+                className ? `AMInstallButton--${className}` : '',
+                {
+                  'AMInstallButton--noDownloadLink': !showDownloadLink,
+                },
+              )}
+              currentVersion={currentVersion}
+              defaultButtonText={defaultButtonText}
+              disabled={!isCompatible}
+              enable={enable}
+              hasAddonManager={hasAddonManager}
+              install={install}
+              isAddonEnabled={isAddonEnabled}
+              puffy={puffy}
+              setCurrentStatus={setCurrentStatus}
+              status={installStatus}
+              uninstall={uninstall}
+            />
+            <GetFirefoxButton
+              addon={addon}
+              buttonType={getFirefoxButtonType}
+              className={className ? `GetFirefoxButton--${className}` : ''}
+            />
+          </>
+        )}
+
         {showDownloadLink ? (
```

Comment:
I still think you can combine this code with the other download link. The text doesn't need to be different. I think "Download file" is appropriate text for both cases, and the fact that the `className` needs to be different can be dealt with programmatically.

Also, the way your patch is now, in some cases there would be two download links, which we don't want. Only having one set of code to render the download link would take care of this.

Reply:
Makes sense. I'll try the ternary operator approach.

Reply:
@bobsilverberg can you direct me to a place where the `{showDownloadLink ? (.....)}` code is being used? I can't seem to figure it out where it is being used. I can't verify my changes unless I know where it is being used. Thanks!

Reply:
It looks like `showDownloadLink` is only defined in this file. It's set on line 95 to `!isCompatible && installURL`, but considering we now have a new criterion for when it should be shown, maybe that can be updated?

Reply:
Yes that was exactly my thought. I'll think of something.

Reply:
I saw that but the whole render of `{showDownloadLink ? (.....)}` is not being used anywhere on the add on versions page, homepage, extensions page. So I was wondering where is it being used. All the buttons on the listed pages are working exactly the same with that whole block commented out.

Reply:
It's the `!isCompatible` that triggers it. So if you go to an add-on page for an add-on that is not compatible with your browser, you should see the download link.

Reply:
Okay thanks! I had the same thought about the link. Let me see what I can do

> It looks like `showDownloadLink` is only defined in this file. It's set on line 95 to `!isCompatible && installURL`, but considering we now have a new criterion for when it should be shown, maybe that can be updated?

Reply:
@bobsilverberg I have made the requested changes i.e. creating a function for rendering the link following DRY.

P.S :- Checked it on Chrome and Firefox to see if only one link is rendering or not and only one is there. I think this is what you wanted? Let me know if any changes are required.